tests/integration/test_charm.py

```diff
@@ -171,10 +128,70 @@ async def test_scale_workload(
     kube_core_client.delete_namespaced_pod(name=leader_pod, namespace=ops_test.model_name)
     await ops_test.model.wait_for_idle(status="active")
 
-    unit_informations = json.loads(
-        (await ops_test.juju("show-unit", app.units[0].name, "--format", "json"))[1]
-    )
-    mattermost_ip = unit_informations[app.units[0].name]["address"]
-    response = requests.get(f"http://{mattermost_ip}:8065", timeout=5)
+    response = requests.get(f"http://{await get_mattermost_ip(ops_test, app)}:8065", timeout=5)
     assert response.status_code == 200
     assert "Mattermost" in response.text
+
+
+async def test_remind_plugin(
+    ops_test: OpsTest,
+    app: Application,
+    test_entities,
+):
+    """
+    arrange: after charm is deployed and ready.
+    act:
+    assert: the plugin should be enabled.
+    """
+
+    plugin_name = "com.github.scottleedavis.mattermost-plugin-remind"
+    cmd = (
+        "MMCTL_LOCAL_SOCKET_PATH=/tmp/mattermost.socket /mattermost/bin/mmctl --local"
+        f" plugin enable {plugin_name}"
+    )
+    await ops_test.juju("run", "--application", app.name, cmd)
```

Comment:
Would this work? https://pythonlibjuju.readthedocs.io/en/latest/api/juju.application.html#juju.application.Application.run

Reply:
Fixed

Reply:
After tests, it seems that I cannot use app.run() successfully. I'm not sure why that's the case (I've tried reading python-libjuju to understand but this code base is hell on earth to me).

Reply:
What issues do you face ?
I remember a case where using the app.run didn't work and had to use juju() but can't remind the use case :/

Reply:
It just doesn't seem to execute the cmd (and with not meaningful output)

Reply:
Conclusion: `app.run()` doesn't work here since we're using podspec on this charm (`app.run()` executes the command in the operator pod instead of the application pod).